plugins/modules/syslog_server.py

```diff
@@ -242,82 +290,189 @@ def update_syslog_server(
     )
     after = new_syserver.to_ansible()  # type: ignore
 
+    records_after = SyslogServer.get_state(rest_client)
     return (
         after != before,
         after,
+        records_after,
         dict(before=before, after=after),
-    )  # changed, records, diff
+    )  # changed, record, records, diff
 
 
 def delete_syslog_server(
-    delete_syserver: SyslogServer, module: AnsibleModule, rest_client: RestClient
-) -> Tuple[bool, Union[TypedSyslogServerToAnsible, Dict[None, None]], TypedDiff]:
+    delete_syserver: Optional[SyslogServer],
+    module: AnsibleModule,
+    rest_client: RestClient,
+) -> Tuple[
+    bool,
+    Union[TypedSyslogServerToAnsible, Dict[None, None]],
+    List[TypedSyslogServerToAnsible],
+    TypedDiff,
+]:
     if not delete_syserver:
-        return False, {}, dict(before={}, after={})
+        records_after = SyslogServer.get_state(rest_client)
+        return False, {}, records_after, dict(before={}, after={})
 
     before = delete_syserver.to_ansible()
     delete_syserver.delete(rest_client, module.check_mode)
 
+    records_after = SyslogServer.get_state(rest_client)
     return (
         True,
         {},
+        records_after,
         dict(before=before, after={}),
     )  # changed, records, diff
 
 
+def _are_equal(
```

Comment:
Should this be an `__eq__ ` method and part of the `syslogserver ` class?

Reply:
Yes. No. Both.

Class `SyslogServer` already has an `__eq__` method, with comment `# This method is here for testing purposes!` - the method returns `True` only if all attributes of `self` and `other` are equal. Also attributes `uuid` and `alert_tag_uuid` are checked. So `True` means not only the configured values are same, but both objects are the same (uuid is same too).

 Here I needed to compare only attributes that can be changed via GUI/ansible module. As many of our modules already have `__eq__` with same "for CI testing only" comment, I decided to leave `__eq__` unmodified.

I will move this to `SyslogServer.is_equivalent`, with comment why this in not `__eq__`.